x-pack/plugins/ml/public/alerting/ml_anomaly_alert_trigger.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useCallback, useEffect, useMemo } from 'react';
+import { EuiSpacer, EuiForm } from '@elastic/eui';
+import { JobSelectorControl } from './job_selector';
+import { useMlKibana } from '../application/contexts/kibana';
+import { jobsApiProvider } from '../application/services/ml_api_service/jobs';
+import { HttpService } from '../application/services/http_service';
+import { SeverityControl } from './severity_control';
+import { ResultTypeSelector } from './result_type_selector';
+import { alertingApiProvider } from '../application/services/ml_api_service/alerting';
+import { PreviewAlertCondition } from './preview_alert_condition';
+import { ANOMALY_THRESHOLD } from '../../common';
+import { MlAnomalyDetectionAlertParams } from '../../common/types/alerts';
+
+interface MlAnomalyAlertTriggerProps {
+  alertParams: MlAnomalyDetectionAlertParams;
+  setAlertParams: <T extends keyof MlAnomalyDetectionAlertParams>(
+    key: T,
+    value: MlAnomalyDetectionAlertParams[T]
+  ) => void;
+  setAlertProperty: (key: string, value: any) => void;
+}
+
+const MlAnomalyAlertTrigger: FC<MlAnomalyAlertTriggerProps> = ({
+  alertParams,
+  setAlertParams,
+  setAlertProperty,
+}) => {
+  const {
+    services: { http },
+  } = useMlKibana();
+  const mlHttpService = useMemo(() => new HttpService(http), [http]);
```

Comment:
this could be moved to a new general `useHttpService` hook in here https://github.com/elastic/kibana/tree/master/x-pack/plugins/ml/public/application/contexts/kibana

Reply:
We already have [useMlApiContext](https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/x-pack/plugins/ml/public/application/contexts/kibana/use_ml_api_context.ts#L11) which is relevant for entire ML plugin. Alerting form lives outside of the ML app context, so I had to create this one. Not sure if it's generic enough